packages/webdriver/src/bidi/core.ts

```diff
@@ -54,18 +55,52 @@ export class BidiCore {
             return
         }
 
-        this.#waitForConnected = new Promise<boolean>((resolve) => {
-            this.#ws.on('open', () => {
-                log.info('Connected session to Bidi protocol')
+        log.info(`Connecting to webSocketUrl ${this.#webSocketUrl}`)
+        const parsedUrl = new URL(this.#webSocketUrl)
+        // https://github.com/webdriverio/webdriverio/issues/14039
+        const candidateUrls: string[] = [this.#webSocketUrl]
+        if (!isIP(parsedUrl.hostname)) {
+            const [ips4, ips6] = await Promise.all([
+                dns.promises.resolve4(parsedUrl.hostname),
+                dns.promises.resolve6(parsedUrl.hostname),
+            ])
+            const candidateIps = [...ips4, ...ips6]
+            candidateUrls.push(
+                ...candidateIps.map((ip) => this.#webSocketUrl.replace(parsedUrl.hostname, ip))
+            )
+        }
+
+        this.#waitForConnected = (async () => {
+            const wsConnectPromises: Promise<WebSocket | null>[] = []
+            for (const candidateUrl of candidateUrls) {
+                const connectPromise = new Promise<WebSocket | null>((resolve) => {
+                    const ws = new environment.value.Socket(candidateUrl, this.#clientOptions) as unknown as WebSocket
+                    ws.once('open', () => resolve(ws))
+                    ws.once('error', (err) => {
+                        const message = `Couldn't connect to Bidi protocol at ${candidateUrl}: ${err.message}`
+                        if (candidateUrls.length > 1) {
+                            log.info(message)
+                        } else {
+                            log.warn(message)
+                        }
+                        resolve(null)
+                    })
+                })
+                wsConnectPromises.push(connectPromise)
+            }
+            const wsMapping = (await Promise.all(wsConnectPromises))
+                .map((ws, index) => [candidateUrls[index], ws])
+                .find(([, ws]) => Boolean(ws)) as [string, WebSocket] | undefined
+            if (wsMapping) {
```

Comment:
not sure if we need to close "extra" opened sockets explicitly here or just rely on the garbage collector